smoot/smoot.py

```diff
@@ -91,14 +102,32 @@ def optimize(self, fun):
             returning y = [ndarray[ne, 1],ndarray[ne, 1],...]
             where y[i][j][0] = fi(xj).
         """
+        if type(self.options["xlimits"]) != np.ndarray:
+            try:
+                self.options["xlimits"] = fun.xlimits
+            except:  # if fun doesn't have "xlimits" attribute
+                print("Error : No bounds given")
+                return
         x_data, y_data = self._setup_optimizer(fun)
-        # n_parallel = self.options["n_parallel"]
-        n_gen = self.options["n_gen"]
-        pop_size = self.options["pop_size"]
-        self.ny = len(y_data)
         self.ndim = self.options["xlimits"].shape[0]
+        # n_parallel = self.options["n_parallel"]
+        try:
+            y_data[0][0][0] += 0
+            self.ny = len(y_data)
+        except:
```

Comment:
I want to catch if y_data is an array/ist (of length n_objectives) of arrays (of size n_points) of arrays( of size 1 containing the value = Fi(xj) ). I do not know how to separate the fact that if y_data has this shape, the it is a MO problem, else if we can only go until y_data[0]|0] to find a value, it means that there is only one objective, so that is the solution that I found

Reply:
So maybe the error to expect is `TypeError` but maybe you can just test for the type with `if isinstance(y_data[0][0], (list, np.ndarray)): ...`

Reply:
good idea I tried this and it worked :
if isinstance(y_data[0][0], float):
            self.log("EGO will be used as there is only 1 objective")
            self.use_ego(fun, x_data, y_data)